contracts/VotingYFI.vy

```diff
@@ -175,21 +185,22 @@ def _checkpoint_user(user: address, old_lock: LockedBalance, new_lock: LockedBal
 
 @internal
 def _checkpoint_global() -> Point:
-    last_point: Point = Point({bias: 0, slope: 0, ts: block.timestamp, blk: block.number})
+    last_point: Point = Point({bias: 0, slope: 0, ts: self.begining_of_week(), blk: block.number})
     epoch: uint256 = self.epoch[self]
     if epoch > 0:
         last_point = self.point_history[self][epoch]
     last_checkpoint: uint256 = last_point.ts
     # initial_last_point is used for extrapolation to calculate block number
     initial_last_point: Point = last_point
     block_slope: uint256 = 0  # dblock/dt
-    if block.timestamp > last_point.ts:
-        block_slope = SCALE * (block.number - last_point.blk) / (block.timestamp - last_point.ts)
+    if self.begining_of_week() > last_point.ts:
+        block_slope = SCALE * (block.number - last_point.blk) / (self.begining_of_week() - last_point.ts)
     
     # apply weekly slope changes and record weekly global snapshots
     t_i: uint256 = self.round_to_week(last_checkpoint)
+    begining_of_week: uint256 = self.begining_of_week()
```

Comment:
Can you declare this a few lines higher so that it can be used on lines 196 and 197?